### PR TITLE
Created DSX Data Structure Definition

### PR DESCRIPTION
**Pull request by jaypacamarra**
The DSX data structure is defined as a bit field struct.
In total, we will use 7+6+1+12+4 = 30 bits out of the 32 bits.

I also defined the initializing function which initializes the values of ID, loc, sign, val, and ret to 0. It will take in a pointer to an object of type DSX_data_t.

### CODE REVIEW

**Comment by manraj-s** [comment]
I think the dsx object in main.c should be a volatile variable or volatile pointer

**Comment by tsaxon** [comment]
Manraj may be onto something, modifying non-volatile dsx_data in a UART ISR _may_ not update the value in the main superloop? But looks good. 1 question though -> what is "DATA_STRUCTURE_INC_DSX_DATA_STRUCTURE_H_" ?

**Comment by jaypacamarra** [comment]
Ya adding volatile will be very good in front of the dsx_data object.

**Comment by jaypacamarra** [comment]
DATA_STRUCTURE_INC_DSX_DATA_STRUCTURE_H_

This is a unique identifier for the include guards of that header file. It can be anything. I can see why that would be confusing because that was auto generated. I can change:

DATA_STRUCTURE_INC_DSX_DATA_STRUCTURE_H_     ->    DSX_DATA_STRUCTURE_H

**Comment by tsaxon** [comment]
I see, messed around with the CubeIDE header templates and it gave me a similar long name. Should we shorten those identifiers for any new libraries were making?

**Comment by jaypacamarra** [comment]
Yes that would be nice if you shorten the default include guard names. Thanks.

File: dsx-fw/Core/Src/main.c [hunk 1]
```diff
@@ -74,7 +74,7 @@ int main(void)
   /* USER CODE BEGIN 1 */
 
 	// Initialize DSX data structure
-	DSX_data_t dsx_data;
+	volatile DSX_data_t dsx_data;
```
Comment (jaypacamarra) [inline dsx-fw/Core/Src/main.c:77]: Added volatile in front so every functions that take in a DSX_data_t must have volatile in it too.

**Comment by jaypacamarra** [comment]
One approval required to merge the pr :)

**Review by manraj-s — APPROVED** [review]
LGTM!

**Pull request merged**